src/Recommendations/Process/Hydration.php

```diff
@@ -73,18 +74,33 @@ public function getPodcastEpisodeChapterById($id): PodcastEpisodeChapterModel
         return new PodcastEpisodeChapterModel($episode, $chapter);
     }
 
+    public function getExternalArticleByOriginalArticleId($id): ExternalArticle
+    {
+        list($originalArticleId, $relatedIndex) = explode('-', $id);
+        $relatedIndex = (int) $relatedIndex;
+
+/** @var ExternalArticle $episode */
+        // TODO: this uses sdk but it should really go through SingleItemRepository (doesn't have a method for this) or in any case through a cache
+        $externalArticle = $this->sdk->getRelatedArticles($originalArticleId)[$relatedIndex];
```

Comment:
indeed, or if their order changes. When an article comes in, we need to prune its existing relations that are not present in the new /related response. Moreover, it could be more resilient to a change in order of the articles, I'll try to use an id rather than an index.